indico_toolkit/indico_wrapper/teach.py

```diff
@@ -0,0 +1,248 @@
+from typing import List
+from functools import wraps
+import pandas as pd
+
+from indico.client.client import IndicoClient
+from indico_toolkit.indico_wrapper import Datasets
+from indico_toolkit.types import TeachTask
+from indico_toolkit import ToolkitInputError
+
+# TODO: Add processor support on duplicate_teach_task
+
+def task_id_required(fn):
+    """
+    'Teach' optionally requires an ID on instantiation. For methods that require that it exists,
+    throw a runtimeError if 'task_id' not set.
+    """
+
+    @wraps(fn)
+    def check_task_id(self, *args, **kwargs):
+        if not isinstance(self.task_id, int):
+            raise RuntimeError(
+                "You must set the 'task_id' attribute to use this method"
+            )
+        return fn(self, *args, **kwargs)
+
+    return check_task_id
+
+
+class Teach(Datasets):
+    """
+    Class to create, delete, and manipulate teach tasks
+    """
+
+    def __init__(self, client: IndicoClient, dataset_id: int, task_id: int = None):
+        super().__init__(client, dataset_id)
+        self.dataset = self.get_dataset()
+        self.task_id = task_id
+        self.df: pd.DataFrame
+
+    def create_teach_task(
```

Comment:
the signature on this is chunky.  feels like this could be wrapped up into a TeachTaskConfig object?